events/views.py

```diff
@@ -9,24 +10,38 @@ class EventIndex(ListView):
     """
 
     model = Event
-    context_object_name = "event_index"
+    context_object_name = "events"
     template_name = "events/index.html"
-    paginate_by = 8
+    paginate_by = settings.PAGINATE_EVENTS_BY
+
+    @property
+    def time_direction(self):
+        if "previous" in self.request.GET:
+            return "previous"
+        elif "upcoming" in self.request.GET:
+            return "upcoming"
+        else:
+            return settings.DEFAULT_TIME_DIRECTION
 
     def get_queryset(self):
         """
         Override get method here to allow us to filter using tags
         """
-        return Event.objects.published(user=self.request.user).order_by("-publish_at")
+        time_direction_mapping = {
+            "previous": Event.objects.past(user=self.request.user).order_by(
+                "-start_at", "-publish_at"
+            ),
+            "upcoming": Event.objects.future(user=self.request.user).order_by(
+                "start_at", "-publish_at"
+            ),
+            "": Event.objects.published(user=self.request.user).order_by("-publish_at"),
```

Comment:
I'd have this as a default variable and return it as the default arg in the .get() call below